packages/web-monetization-demo/src/server/SPSP.ts

```diff
@@ -2,42 +2,62 @@ import {
   BaseHttpController,
   controller,
   httpGet,
+  httpPost,
   requestHeaders,
+  requestBody,
   requestParam,
   response
 } from 'inversify-express-utils'
 import * as express from 'express'
 
+import { ReceiptVerifier } from './ReceiptVerifier'
 import { StreamServer } from './StreamServer'
 import { dbg } from './logging'
 
 @controller('/')
 export class SPSP extends BaseHttpController {
-  constructor(private streamServer: StreamServer) {
+  constructor(
+    private streamServer: StreamServer,
+    private receiptVerifier: ReceiptVerifier
+  ) {
     super()
   }
 
   @httpGet('spsp/:pointer')
   async spsp(
-    @requestHeaders('web-monetization-id') requestId: string,
+    @requestHeaders('receipt-nonce') receiptNonce: string,
```

Comment:
These aren't actually used anywhere here right

Reply:
It might actually be nice to separate the Verifier and Receiver into separate services (i.e. running on separate ports) ? 

Just throwing it out as an idea :)